test/utils/test_get_openai_tool_schema.py

```diff
@@ -0,0 +1,230 @@
+                        'description': 'int_para desc'
+                    },
+                    'list_para': {
+                        'items': {
+                            'type': 'integer'
+                        },
+                        'type': 'array',
+                        'description': 'list_para desc'
+                    },
+                    'float_para': {
+                        'type': 'number',
+                        'description': 'float_para desc'
+                    },
+                    'datatime_para': {
+                        'format': 'date-time',
+                        'type': 'string',
+                        'description': 'datatime_para desc'
+                    },
+                    'default_enum_para': {
+                        'allOf': [{
+                            '$ref': '#/$defs/RoleType'
+                        }],
+                        'default': 'critic',
+                        'description': 'default_enum_para desc'
+                    }
+                },
+                'required': [
+                    'str_para', 'int_para', 'list_para', 'float_para',
+                    'datatime_para'
+                ],
+                'type':
+                'object'
+            }
+        }
+    }
+
+    openai_tool_schema = get_openai_tool_schema(test_all_parameters)
+
+    assert openai_tool_schema == expect_res
+
+    client = OpenAI()
+    expect_res = json.loads("""{
+      "str_para": "hello",
+      "int_para": 1,
+      "list_para": [1, 2],
+      "float_para": 0.1,
+      "datatime_para": "2023-11-21T15:28:12.041872",
+      "default_enum_para": "critic"
+    }""")
+    tool_response = client.chat.completions.create(
+        model="gpt-3.5-turbo", messages=[{
+            "role":
+            "user",
+            "content":
+            "Call the function "
+            "test_all_parameter with: "
+            "str_para:hello,int_para:1,list_para:[1,2],"
+            "float_para:0.1,"
+            "datatime_para:2023-11-21T15:28:12.041872,"
+            "default_enum_para:RoleType.CRITIC"
+        }], tools=[get_openai_tool_schema(test_all_parameters)])
+    assert (tool_response.choices[0].finish_reason == "tool_calls")
+    tool_params = json.loads(
+        tool_response.choices[0].message.tool_calls[0].function.arguments)
+    assert tool_params == expect_res
+    function_response = client.chat.completions.create(
+        model="gpt-3.5-turbo", messages=[{
+            "role":
+            "user",
+            "content":
+            "Call the function test_all_parameter with:"
+            " str_para:hello,"
+            "int_para:1,"
+            "list_para:[1,2],"
+            "float_para:0.1,"
+            "datatime_para:2023-11-21T15:28:12.041872,"
+            "default_enum_para:RoleType.CRITIC"
+        }], functions=[get_openai_function_schema(test_all_parameters)])
+    assert function_response.choices[0].finish_reason == "function_call"
```

Comment:
`function_call` is deprecated which may be removed later. Can we just remove the support for `get_openai_function_schema`?

Reply:
Ahh,we are using it in the chat agent. So let's keep it for now and remove it in another PR.